app/src/main/java/com/appdev/all_in_android/ui/screens/MarketPlaceScreen.kt

```diff
@@ -1,12 +1,132 @@
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SearchBar
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.appdev.all_in_android.R
+import com.appdev.all_in_android.ui.components.general.AllInTopBar
+import com.appdev.all_in_android.ui.components.general.PlayerCard
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun MarketplaceScreen(){
-    Box(){
-        Text(text = "marketplace")
+fun MarketPlaceScreen() {
+    Scaffold(
+        topBar = { AllInTopBar("Marketplace", 1000) }
+    ) { padding ->
+        var searchQuery by remember { mutableStateOf("") }
+
+        Column(
+            modifier = Modifier
+                .padding(padding)
+                .background(color = Color(0xffE6E6E6))
+        ){
+            Column(modifier = Modifier
+                .padding(16.dp)
+                .fillMaxSize()
+
+            ){
+                SearchBar(
+                    searchQuery,
+                    { searchQuery = it }
+                )
+                FilterRow()
+
+                LazyColumn{
+                    items(count = 5){
+                        Row(
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            modifier = Modifier.padding(top = 8.dp, bottom = 8.dp)
+                        ){
+                            PlayerCard(R.drawable.player_photo, "Player Name", "04/26", 4, "FGA", 1000, 2000)
+                            Spacer(Modifier.padding(2.dp))
+                            PlayerCard(R.drawable.player_photo, "Player Name", "04/26", 4, "FGA", 1000, 2000)
+                        }
+                    }
+                }
+            }
+        }
     }
+}
+
+@Composable
+fun SearchBar(
+    query: String,
+    onQueryChanged: (String) -> Unit
+){
+    OutlinedTextField(
+        value = query,
+        onValueChange = onQueryChanged,
+        shape = RoundedCornerShape(16.dp),
+        modifier = Modifier.fillMaxWidth()
+            .clip(RoundedCornerShape(16.dp))
+            .background(color = Color.White, shape = RoundedCornerShape(16.dp)),
+        placeholder = { Text("Search...") },
+        leadingIcon = {
+            Icon(
+                imageVector = Icons.Default.Search,
+                contentDescription = "Search icon"
+            )
+        },
+        singleLine = true
+    )
+}
+
+@Composable
+fun FilterRow(){
```

Comment:
I think it would be better to pass in a list of filters as a parameter for FilterRow.